File: Fundamental courses of the major/Math For Computer Science/TAI LIEU KHOA TRUOC/C++ & python/C1 Number theory/Nearest prime.py
import math

def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def nearest_prime(N):
    lower_prime = N
    while lower_prime > 1 and not is_prime(lower_prime):
        lower_prime -= 1

    upper_prime = N
    while not is_prime(upper_prime):
        upper_prime += 1

    if N - lower_prime <= upper_prime - N:
        return lower_prime
    else:
        return upper_prime
    
def main():
    N = int(input("Nhập số N: "))
    closest_prime = nearest_prime(N)
    print(f"Số nguyên tố gần nhất với {N} là: {closest_prime}")
main()

File: Fundamental courses of the major/Math For Computer Science/TAI LIEU KHOA TRUOC/C++ & python/C1 Number theory/Nearest prime.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool is_prime(int n) {
    if (n <= 1) return false;
    for (int i = 2; i <= sqrt(n); ++i) {
        if (n % i == 0) return false;
    }
    return true;
}

int nearest_prime(int N) {
    int lower_prime = N;
    while (lower_prime > 1 && !is_prime(lower_prime)) {
        --lower_prime;
    }

    int upper_prime = N;
    while (!is_prime(upper_prime)) {
        ++upper_prime;
    }

    if (N - lower_prime <= upper_prime - N) {
        return lower_prime;
    } else {
        return upper_prime;
    }
}

int main() {
    int N;
    cout << "Nhap so N: ";
    cin >> N;

    int closest_prime = nearest_prime(N);
    cout << "So nguyen to gan nhat voi " << N << " la: " << closest_prime << endl;

    return 0;
}
